processor/src/operations/comb_ops.rs

```diff
@@ -0,0 +1,267 @@
+// RANDOM LINEAR COMBINATION OPERATIONS
+// ================================================================================================
+
+use vm_core::{Felt, Operation, StarkField, ONE, ZERO};
+
+use crate::{ExecutionError, Host, Process, QuadFelt};
+
+impl<H> Process<H>
+where
+    H: Host,
+{
+    // COMBINE VALUES USING RANDOMNESS
+    // --------------------------------------------------------------------------------------------
+    /// Performs a single step in the computation of the random linear combination:
+    ///
+    /// \sum_{i=0}^k{\alpha_i \cdot \left(\frac{T_i(x) - T_i(z)}{x - z} +
+    ///            \frac{T_i(x) - T_i(g \cdot z)}{x - g \cdot z} \right)}
+    ///
+    /// The instruction computes the numerators $\alpha_i \cdot (T_i(x) - T_i(z))$ and
+    /// $\alpha_i \cdot (T_i(x) - T_i(g \cdot z))$ and stores the values in two accumulators $p$
+    /// and $r$, respectively. This instruction is specialized to main trace columns i.e.
+    /// the values $T_i(x)$ are base field elements.
+    /// The instruction is used in the context of STARK proof verification in order to compute
+    /// the queries of the DEEP composition polynomial for FRI. It works in combination with
+    /// the `mem_stream` instruction where it is called 8 times in a row for each call to
+    /// `mem_stream`.
+    ///
+    /// The stack transition of the instruction can be visualized as follows:
+    ///
+    /// Input:
+    ///
+    /// +------+------+------+------+------+------+------+------+------+------+------+------+------+------+------+---+
+    /// |  T7  |  T6  |  T5  |  T4  |  T3  |  T2  |  T1  |  T0  |  p1  |  p0  |  r1  |  r0  |x_addr|z_addr|a_addr| - |
+    /// +------+------+------+------+------+------+------+------+------+------+------+------+------+------+------+---+
+    ///
+    ///
+    /// Output:                                                      
+    ///                                                    
+    /// +------+------+------+------+------+------+------+------+------+------+------+------+------+--------+--------+---+
+    /// |  T0  |  T7  |  T6  |  T5  |  T4  |  T3  |  T2  |  T1  |  p1' |  p0' |  r1' |  r0' |x_addr|z_addr+1|a_addr+1| - |
+    /// +------+------+------+------+------+------+------+------+------+------+------+------+------+--------+--------+---+
+    ///
+    ///
+    /// Here:
+    ///
+    /// 1. Ti for i in 0..=7 stands for the the value of the i-th trace polynomial for the current
+    ///  query i.e. T_i(x).
+    /// 2. (r0, r1) stands for an extension field element accumulating the values for the quotients
+    ///  with common denominator (x - gz).
+    /// 3. (p0, p1) stands for an extension field element accumulating the values for the quotients
+    ///  with common denominator (x - z).
```

Comment:
tiny nit: I'd flip the order of these bullet points so that they are in the same order as in mdBook docs (i.e., `2. (p0, p1) ...` and `3. (r0, r1) ... `

Reply:
Agreed and changed